azurelinuxagent/common/protocol/hostplugin.py

```diff
@@ -202,7 +217,29 @@ def should_report(is_healthy, error_state, last_timestamp, period):
         return datetime.datetime.utcnow() >= (last_timestamp + period)
 
     def put_vm_log(self, content):
-        raise NotImplementedError("Unimplemented")
+        """
+        Try to upload VM logs, a compressed zip file, via the host plugin /vmAgentLog channel.
+        :param content: the binary content of the zip file to upload
+        """
+        if not self.ensure_initialized():
+            raise ProtocolError("HostGAPlugin: HostGAPlugin is not available")
+
+        if content is None or self.container_id is None or self.deployment_id is None:
```

Comment:
Not sure if you already investigated this, but we should pick up containerId (and maybe deploymentId) from AgentGlobals class instead as that data is always up-to-date

Reply:
Or do we handle that externally (outside this class)?